Broker/SqsFactory.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Swarrot\SwarrotBundle\Broker;
+
+use Aws\Sqs\SqsClient;
+use Swarrot\MessageProvider\SqsMessageProvider;
+use Swarrot\Broker\MessageProvider\MessageProviderInterface;
+use Swarrot\Broker\MessagePublisher\MessagePublisherInterface;
+use Swarrot\SwarrotBundle\Broker\FactoryInterface;
+
+/**
+ * Class SqsFactory
+ */
+class SqsFactory implements FactoryInterface
+{
+    protected $connections = array();
+    protected $messageProviders = array();
+    protected $messagePublishers = array();
+
+    /**
+     * {@inheritDoc}
+     */
+    public function addConnection($name, array $connection)
+    {
+        $this->connections[$name] = $connection;
+    }
+
+    /**
+     * getMessageProvider.
+     *
+     * @param string $name       The name of the queue where the MessageProviderInterface will found messages
+     * @param string $connection The name of the connection to use
+     *
+     * @return MessageProviderInterface
+     */
+    public function getMessageProvider($name, $connection)
+    {
+        if (!isset($this->messageProviders[$connection][$name])) {
+            if (!isset($this->messageProviders[$connection])) {
+                $this->messageProviders[$connection] = array();
+            }
+
+            $channel = $this->getChannel($connection);
+
+            $this->messageProviders[$connection][$name] = new SqsMessageProvider($channel, $this->connections[$connection]['vhost'] . $name);
+        }
+
+        return $this->messageProviders[$connection][$name];
+    }
+
+    /**
+     * getMessagePublisher.
+     *
+     * @param string $name       The name of the exchange where the MessagePublisher will publish
+     * @param string $connection The name of the connection to use
+     *
+     * @return MessagePublisherInterface
+     */
+    public function getMessagePublisher($name, $connection)
+    {
+        throw new \Exception('Implement method getMessagePublisher');
```

Comment:
``BadMethodCallException('Publishing messages to SQS is not implemented yet')`` would be better (and btw, it would be great to implement it)